httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetBalancesTest.kt

```diff
@@ -0,0 +1,252 @@
+        }
+        install(Auth) {
+          bearer {
+            sendWithoutRequest { true }
+          }
+        }
+      }
+    }
+
+    @Test
+    fun `returns 401 if no Bearer token is present`() = runBlocking {
+
+      val response = client.get("/balances")
+
+      assertEquals(HttpStatusCode.Unauthorized, response.status)
+
+      val errorResponse = Json.jsonMapper.readValue(response.bodyAsText(), ErrorResponse::class.java)
+      assertContains(errorResponse.errors.first().detail, "Authorization header required")
+    }
+
+    @Test
+    fun `returns 401 if malformed Bearer token is present`() = runBlocking {
+
+      val response = client.get("/balances") {
+        bearerAuth("Bearer abc123")
+      }
+
+      assertEquals(HttpStatusCode.Unauthorized, response.status)
+
+      val errorResponse = Json.jsonMapper.readValue(response.bodyAsText(), ErrorResponse::class.java)
+      assertContains(errorResponse.errors.first().detail, "Malformed Authorization header.")
+    }
+
+    @Test
+    fun `returns 200 if balances are found`() = runBlocking {
+      val balance = Balance.create(
+        from = TestData.pfiDid.uri,
+        data = BalanceData(
+          currencyCode = "USD",
+          available = "100.00"
+        )
+      )
+      balance.sign(TestData.pfiDid)
+      balancesApi.addBalance(balance)
+      val response = client.get("/balances") {
+        bearerAuth(RequestToken.generate(TestData.aliceDid, TestData.pfiDid.uri))
+      }
+
+      assertEquals(HttpStatusCode.OK, response.status)
+
+      val responseString = response.bodyAsText()
+      val jsonNode = Json.jsonMapper.readTree(responseString)
+      val balances = jsonNode.get("data").elements().asSequence()
+        .map { balance ->
+          Resource.parse(balance.toString())
+        }
+        .toList()
+
+      assertEquals(1, balances.size)
+    }
+
+  }
+
+  @Nested
+  inner class GetBalancesMockkTest {
+    private lateinit var applicationCall: ApplicationCall
+    private lateinit var balancesApi: BalancesApi
+    private val callback: GetBalancesCallback = mockk(relaxed = true)
+    private val aliceDid = DidDht.create(InMemoryKeyManager())
+
+    @BeforeEach
+    fun setUp() {
+      applicationCall = mockk(relaxed = true)
+      balancesApi = mockk(relaxed = true)
+    }
+
+    @Test
+    fun `verify callback is invoked upon successful get balances`() = runBlocking {
+      val balances = listOf<Balance>(mockk(relaxed = true))
+      coEvery { applicationCall.request.headers[HttpHeaders.Authorization] } returns
+        "Bearer ${
+          RequestToken.generate(
+            aliceDid,
+            "did:ex:pfi"
+          )
+        }"
+      coEvery { balancesApi.getBalances(any<String>()) } returns balances
+
+      getBalances(applicationCall, balancesApi, callback, "did:ex:pfi")
+
+      coVerify(exactly = 1) { callback.invoke(applicationCall) }
+      coVerify { applicationCall.respond(HttpStatusCode.OK, any<GetBalancesResponse>()) }
```

Comment:
niceeee. thanks for setting this up. it'll be easy now to copy this pattern for other callback tests in this library.